### PR TITLE
Cleanup docs of bezier

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/bezier.py [hunk 1]
```diff
@@ -99,21 +99,41 @@ def split_de_casteljau(beta, t):
...
+    doesn't. The search stops when the distance of the points parametrized by
+    *t0* and *t1* gets smaller than the given *tolerance*.
 
-    """
-    # inside_closedpath : function
+    Parameters
+    ----------
+    bezier_point_at_t : callable
+        A function returning x, y coordinates of the bezier at parameter *t*.
+        it must have the signature::
```
Comment (anntzer) [inline lib/matplotlib/bezier.py]: capitalize "it"

File: lib/matplotlib/bezier.py [hunk 2]
```diff
@@ -99,21 +99,41 @@ def split_de_casteljau(beta, t):
...
         bezier_point_at_t, inside_closedpath, t0=0., t1=1., tolerance=0.01):
     """
-    Find a parameter t0 and t1 of the given bezier path which
-    bounds the intersecting points with a provided closed
-    path(*inside_closedpath*). Search starts from *t0* and *t1* and it
-    uses a simple bisecting algorithm therefore one of the end point
-    must be inside the path while the other doesn't. The search stop
-    when |t0-t1| gets smaller than the given tolerance.
-    value for
+    Find the intersection of the bezier curve with a closed path.
```
Comment (anntzer) [inline lib/matplotlib/bezier.py]: Capitalize Bezier throughout.

File: lib/matplotlib/bezier.py [hunk 3]
```diff
@@ -147,21 +167,22 @@ def find_bezier_t_intersecting_with_closedpath(
...
 class BezierSegment:
     """
-    A simple class of a 2-dimensional bezier segment
+    A 2-dimensional bezier segment.
+
+    Parameters
+    ----------
+    control_points : array-like (N, 2)
+        A list of the (x, y) positions of contol points of the bezier line.
+        This must contain N points ,where N is the order of the bezier line.
```
Comment (anntzer) [inline lib/matplotlib/bezier.py]: spacing around comma

File: lib/matplotlib/bezier.py [hunk 4]
```diff
@@ -147,21 +167,22 @@ def find_bezier_t_intersecting_with_closedpath(
...
+        1<= N <= 3 is supported.
```
Comment (anntzer) [inline lib/matplotlib/bezier.py]: spacing after 1

File: lib/matplotlib/bezier.py [hunk 5]
```diff
@@ -309,8 +357,22 @@ def get_cos_sin(x0, y0, x1, y1):
...
-    Return 1 if two lines are parallel in same direction, -1 if two lines are
-    parallel in opposite direction, 0 otherwise.
+    Check if two lines are parallel.
+
+    Parameters
+    ----------
+    dx1, dy1, dx2, dy2 : float
+        The gradients *dy*/*dx* of the two lines.
+    tolerance : float
+        The angular tolerange in rad up to which the lines are considered
```
Comment (anntzer) [inline lib/matplotlib/bezier.py]: radians

**Comment by anntzer** [comment]
see also #13355 :)

File: lib/matplotlib/bezier.py [hunk 6]
```diff
@@ -99,21 +99,41 @@ def split_de_casteljau(beta, t):
...
-    when |t0-t1| gets smaller than the given tolerance.
-    value for
+    Find the intersection of the Bezier curve with a closed path.
 
-    - bezier_point_at_t : a function which returns x, y coordinates at *t*
+    The intersection point *t* is approximated by two parameters *t0*, *t1*
+    such that *t0* <= *t* <= *t1*.
 
-    - inside_closedpath : return True if the point is inside the path
+    Search starts from *t0* and *t1* and it uses a simple bisecting algorithm
```
Comment (QuLogic) [inline lib/matplotlib/bezier.py]:
```suggestion
    Search starts from *t0* and *t1* and uses a simple bisecting algorithm,
```

File: lib/matplotlib/bezier.py [hunk 7]
```diff
@@ -309,8 +357,22 @@ def get_cos_sin(x0, y0, x1, y1):
...
-    Return 1 if two lines are parallel in same direction, -1 if two lines are
-    parallel in opposite direction, 0 otherwise.
+    Check if two lines are parallel.
+
+    Parameters
+    ----------
+    dx1, dy1, dx2, dy2 : float
+        The gradients *dy*/*dx* of the two lines.
+    tolerance : float
+        The angular tolerange in radians up to which the lines are considered
```
Comment (QuLogic) [inline lib/matplotlib/bezier.py]:
```suggestion
        The angular tolerance in radians up to which the lines are considered
```

**Comment by NelleV** [comment]
Thanks @timhoffm !

**Pull request merged**